src/pages/index.js

```diff
@@ -5,52 +5,348 @@ import PopupWithForm from "../components/PopupWithForm.js";
+        items: cardData,
+        renderer: (cardData) => {
+          renderCard(cardData);
+        },
+      },
+      ".cards__list"
+    );
+    section.renderItems();
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
+// api
+//   .getInitialCards()
+//   .then((cardData) => {
+//     if (cardData) {
+//       section = new Section(
+//         {
+//           items: cardData,
+//           renderer: renderCard,
+//         },
+//         cardListEl
+//       );
+//       section.renderItems();
+//     }
+//   })
+//   .catch((err) => console.log("Error loading cards:", err));
+
+let user;
+
+api
+  .getUserInfo()
+  .then((userData) => {
+    userInfo.setUserInfo(userData);
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
+// avatar edit modal
+
+const avatarChangeModal = new PopupWithForm(
+  "#avatar-modal",
+  handleAvatarChangeSubmit
+);
+avatarChangeModal.setEventListeners();
+
+// ---------------------------------------
+const avatarSubmitButton = document.querySelector("#avatar-submit-button");
+
+function handleAvatarChangeSubmit(inputValues) {
+  // event.preventDefault();
+  avatarSubmitButton.textContent = "Saving...";
+
+  // const avatarForm = event.currentTarget;
+  const avatarUrl = inputValues.avatar;
+  console.log("avatarUrl:", avatarUrl);
+  // const avatarUrl = event.target.avatar.value;
+
+  if (avatarUrl) {
+    api
+      .updateAvatar(avatarUrl)
+      .then((updatedUser) => {
+        userInfo.setUserInfo(updatedUser);
+        avatarChangeModal.close();
+      })
+      .catch((err) => console.error("Error updating avatar:", err))
+      .finally(() => {
+        avatarSubmitButton.textContent = "Save";
+      });
+  } else {
+    console.error("Avatar URL is not defined");
+    avatarSubmitButton.textContent = "Save"; // Ensure the button text is reset
+  }
+}
+
+const avatarFormElement = document.querySelector("#avatar-form");
+if (avatarFormElement) {
+  avatarFormElement.addEventListener("submit", (event) => {
+    event.preventDefault();
+    const inputValues = new FormData(event.target);
+    handleAvatarChangeSubmit(Object.fromEntries(inputValues));
+  });
+}
+
+// Event Listener for opening avatar modal
+avatarImage.addEventListener("click", () => {
+  submitButtonSelector.textContent = "Save";
+  avatarChangeModal.open();
+  avatarFormValidator.toggleButtonState();
+});
+console.log("User object:", user);
+
+// ----------------------------
+
+// Delete card modal
+
+const deleteModal = document.querySelector("#delete-modal");
+const deleteModalCloseButton = document.querySelector("#delete-close-modal");
+const deleteConfirmForm = document.querySelector("#delete-confirm-form");
+const deleteConfirmModal = new ConfirmPopup(
+  "#delete-modal",
+  handleCardDeleteSubmit
+);
+deleteConfirmModal.setEventListeners();
+
+let cardToDelete = null;
+
+// Add event listeners to open the modal when delete button is clicked
+document.querySelectorAll(".card__delete-button").forEach((button) => {
```

Comment:
Please, delete it


Class `Card` should be responsible only for creating a card and adding 3 listeneres: like, deleting and clicks on images

Now only class `Card` knows every **card selector**, so you need to encapsulate the deleting logics inside its methods and call them in `index.js`. It’s a violation of **Single responsibility** principle

Reply:
so i should move all the delete like and clicks in card.js and just import it in index.js?

Reply:
> so i should move all the delete like and clicks in card.js and just import it in index.js?

It's done inside class `Card` (not in `index.js`)